DynamicSmoothedGUI.py

```diff
@@ -96,57 +99,88 @@ def create_widgets(self, empty=True):
-                            curr_rec, outline="#ff621f", fill="#ff621f")
-                    elif (curr_tile.isBloated):
-                        self.canvas.itemconfig(
-                            curr_rec, outline="#ffc0cb", fill="#ffc0cb")
-                    elif (curr_tile in self.visitedSet):
-                        self.canvas.itemconfig(
-                            curr_rec, outline="#0C9F34", fill="#0C9F34")
-                    elif (curr_tile not in self.visitedSet):
-                        self.canvas.itemconfig(
-                            curr_rec, outline="#fff", fill="#fff")
-                else:
-                    if (
-                            curr_tile.isObstacle == False and curr_tile.isBloated == False and curr_tile not in self.visitedSet):
-                        self.canvas.itemconfig(
-                            curr_rec, outline="#545454", fill="#545454")
+        index_radius_inner = int(vis_radius / tile_size)
+        index_radius_outer = index_radius_inner + 2
+
+        # the bounds for the visibility circle
+        lower_row = int(max(0, row - index_radius_outer))
+        lower_col = int(max(0, col - index_radius_outer))
+        upper_row = int(min(row + index_radius_outer, self.gridFull.num_rows))
+        upper_col = int(min(col + index_radius_outer, self.gridFull.num_cols))
+
+        lidar_data = self.generate_sensor.generateLidar(degree_freq, row, col)
+        rad_inc = int(GUI_tile_size / 3)  # radius increment to traverse tiles
+
+        def _color_normally(coords):
+            """
+            Colors the tile at coordinates based on its known attribute (obstacle, bloated, visited, or seen).
+            (coordinates must be in bounds of GUI window)
+            """
+            curr_tile = self.gridEmpty.grid[int(coords[0])][int(coords[1])]
+            curr_rec = self.tile_dict[curr_tile]
+            if not curr_tile.isKnown:
+                # tile has not been "seen" before; i.e. no information is known about this tile
+                if curr_tile.isObstacle and not curr_tile.isBloated:
+                    self.canvas.itemconfig(curr_rec, outline="#ff621f", fill="#ff621f")  # red
+                elif curr_tile.isBloated:
+                    self.canvas.itemconfig(curr_rec, outline="#ffc0cb", fill="#ffc0cb")  # pink
+                else:  # available path in range of sight
+                    self.canvas.itemconfig(curr_rec, outline="#fff", fill="#fff")  # white
+                curr_tile.isKnown = True
+            elif curr_tile in self.visitedSet:  # tile on path already travelled
+                self.canvas.itemconfig(curr_rec, outline="#0C9F34", fill="#0C9F34")  # green
+
+        for deg in range(0, 360, degree_freq):
+            # iterating through 360 degrees surroundings of robot in increments of degree_freq
+            angle_rad = deg * math.pi / 180
+            if len(lidar_data) == 0 or (len(lidar_data) != 0 and lidar_data[0][0] != deg):
```

Comment:
Get rid of len(lidar_data)!=0 condition because we know that this is true... if len((lidar_data) == 0 then the first part of the OR condition would be satisficed and the Boolean condition would short circuit and evaluate to true